subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/FileSystemWatchingIntegrationTest.groovy

```diff
@@ -794,6 +795,102 @@ class FileSystemWatchingIntegrationTest extends AbstractIntegrationSpec implemen
         "symlink in a directory"        | "dirWithSymlink/symlinkInside" | "fileInside.txt" | 'dir("dirWithSymlink")' | "fileInside.txt"
     }
 
+    @Requires(TestPrecondition.SYMLINKS)
+    def "file system watching works the project dir is symlinked"() {
+        def actualProjectDir = file("parent/projectDir")
+        def symlink = file("symlinkedParent")
+        symlink.createLink(file("parent"))
+
+        def fileToChange = actualProjectDir.file("actualFile")
+        fileToChange.createFile()
+
+        actualProjectDir.file("build.gradle") << """
+            task myTask {
+                def outputFile = file("build/output.txt")
+                inputs.file("actualFile")
+                outputs.file(outputFile)
+
+                doLast {
+                    outputFile.text = "Hello world"
+                }
+            }
+        """
+        actualProjectDir.file("settings.gradle").createFile()
+        executer.beforeExecute {
+            // Use `new File` here to avoid canonicalization of the path
+            def symlinkedProjectDir = new File(symlink, "projectDir")
+            assert !symlinkedProjectDir.absolutePath.startsWith(actualProjectDir.absolutePath)
```

Comment:
Why check for `.startsWith()` and not `.equals()`?

Reply:
I switched to using `equals/==`.